megamek/src/megamek/server/GameManager.java

```diff
@@ -6960,6 +6960,18 @@ private void processMovement(Entity entity, MovePath md, Map<EntityTargetPair,
                 }
             }
 
+            // check for tank fortify
+            if (entity instanceof Tank) {
+                Tank tnk = (Tank) entity;
+                if (step.getType() == MovePath.MoveStepType.FORTIFY) {
+                    if (!tnk.hasWorkingMisc(MiscType.F_BULLDOZER)
+                            && !tnk.hasWorkingMisc("Backhoe")) {
+                        sendServerChat(entity.getDisplayName()
+                                + " failed to fortify because it is missing suitable equipment");
+                    }
+                    tnk.setDugIn(Tank.DUG_IN_FORTIFYING1);
```

Comment:
Is this missing a return statement after the error chat message?

Reply:
The original code did not have it so I don't know. Also, there is another spot like this from the original code on line 6955 and several others higher up.

Reply:
Well looking at it, certainly not a return. It's strange that when the unit somehow hasn't the gear anymore it would send a message but then set the unit to FORTIFYING1 all the same. I think the problem is that this never happens with infantry, how should they lose their shovels so it's never tested. Problem for another day.

Reply:
The original dig in and fortify code is from 2006, who knows why it was written that way...